packages/lib/src/index.ts

```diff
@@ -72,36 +78,74 @@ export enum EventType {
   Error = "error"
 }
 
-export type RequestId = string
+export type RequestID = string
+
+export type EventListenerHandler<T> = (
+  event: EventType,
+  data: T | ErrorCode
+) => void
+
+export const VALID_DOMAIN = "https://windowai.io" as const
+
+export interface WindowAI<TModel = string> {
+  __window_ai_metadata__: {
+    domain: typeof VALID_DOMAIN
+    version: string
+  }
+
+  getCompletion(
+    input: Input,
+    options?: CompletionOptions<TModel>
+  ): Promise<Output | Output[]>
+
+  getCurrentModel(): Promise<TModel>
+
+  addEventListener<T>(handler: EventListenerHandler<T>): RequestID
+}
 
 declare global {
   interface Window {
-    ai: {
-      getCompletion(
-        input: Input,
-        options: CompletionOptions
-      ): Promise<Output | Output[]>
-
-      getCurrentModel(): Promise<ModelID>
-      addEventListener<T>(
-        handler: (event: EventType, data: T | ErrorCode) => void
-      ): RequestId
-    }
+    ai: WindowAI
   }
 }
 
-export const initWindowAI = async () => {
-  // wait until the window.ai object is available
-  if (!window.ai) {
-    await new Promise((resolve) => {
-      const interval = setInterval(() => {
-        if (window.ai) {
-          clearInterval(interval)
-          resolve(1)
-        }
-      }, 100)
-    })
+// Checking against other window.ai implementations
+export function hasWindowAI() {
+  return (
+    !!globalThis.window.ai?.__window_ai_metadata__ &&
```

Comment:
Unfortunately, we announced the extension and pointed people to download it prior to injecting this metadata into the window.ai object, so this is going to return false for all the prior users.

Open to other ideas here, but I think we should just check for the existence of `window.ai?.getCompletion` for now, and switch to this code down the road. (we could leave it commented here with a TODO)